### PR TITLE
misc: use re2 if configured

### PR DESCRIPTION
**Pull request by mhoffm-aiven**
Make it possible to use re2 as a non-backtracking engine to evaluate searches. We make it opt-in, configurable using environment since otherwise it would constitute a breaking change.

### CODE REVIEW

**Review by aiven-amartin — APPROVED** [review]
LGTM, @mhoffm-aiven if you can update the description a bit with context, will hold on a second reviewer before merging. [SRE-8038]

**Review by aiven-amartin — APPROVED** [review]

**Review by exg — APPROVED** [review]

**Pull request merged**
